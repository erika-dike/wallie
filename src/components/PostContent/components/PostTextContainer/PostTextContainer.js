import React from 'react';
import PropTypes from 'prop-types';


const PostTextContainer = ({ text }) => {
  const createMarkup = () => ({ __html: `${text}` });

  return (
    <div className="post-text-container">
      <div className="post-text-size post-text">
        <div dangerouslySetInnerHTML={createMarkup()} />
      </div>
    </div>
  );
};

PostTextContainer.propTypes = {
  text: PropTypes.string.isRequired,
};

export default PostTextContainer;
