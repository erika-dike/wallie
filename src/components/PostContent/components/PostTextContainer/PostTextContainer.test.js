import React from 'react';
import renderer from 'react-test-renderer';

import PostTextContainer from './PostTextContainer';


it('renders correctly', () => {
  const tree = renderer.create(
    <PostTextContainer text="There are always biger fishes to fry" />,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
