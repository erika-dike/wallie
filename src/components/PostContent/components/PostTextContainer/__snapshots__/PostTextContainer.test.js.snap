exports[`test renders correctly 1`] = `
<div
  className="post-text-container">
  <div
    className="post-text-size post-text">
    <div
      dangerouslySetInnerHTML={
        Object {
          "__html": "There are always biger fishes to fry",
        }
      } />
  </div>
</div>
`;
